pkg/cybr/conjur/authenticators/authenticators.go

```diff
@@ -0,0 +1,34 @@
+package authenticators
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/infamousjoeg/cybr-cli/pkg/cybr/conjur/authenticators/iam"
+	"github.com/infamousjoeg/cybr-cli/pkg/cybr/helpers/authenticators"
+)
+
+func getAuthenticators(config authenticators.Config) ([]authenticators.Authenticator, error) {
+	authenticators := []authenticators.Authenticator{}
+	iamInterface, err := iam.New()
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create IAM authenticator. %s", err)
+	}
+	authenticators = append(authenticators, iamInterface)
+	return authenticators, nil
+}
```

Comment:
I am slightly confused regarding how api key authentication is implemented here.

Reply:
If you look at [cmd/conjur.go] L119, if the `logon` command is used with a username and password, then the `authn` method will be utilized with Conjur and written to .conjurrc and .netrc.  If environment variables are set and no .conjurrc or .netrc is present, `authn-iam` will be utilized and that function will get used.

```
CONJUR_ACCOUNT=cyberarkdemo CONJUR_APPLIANCE_URL=https://conjur.joegarcia.dev CONJUR_AUTHN_LOGIN=host/cloud/aws/ec2/735280068473/ConjurAWSRoleEC2 CONJUR_AUTHENTICATOR=authn-iam CONJUR_AUTHN_SERVICE_ID=prod CONJUR_AWS_TYPE=ec2 cybr conjur list
[
    "cyberarkdemo:variable:cloud/aws/ec2/psql/db01/address",
    "cyberarkdemo:variable:cloud/aws/ec2/psql/db01/username",
    "cyberarkdemo:variable:cloud/aws/ec2/psql/db01/password",
    "cyberarkdemo:variable:cloud/aws/ec2/psql/db01/db-name",
    "cyberarkdemo:webservice:conjur/authn-iam/prod"
]
```